content/develop/start/truffle.md

```diff
@@ -1,5 +1,224 @@
+
+Now you can test the flow as mentioned in the [NFT Example](#nft-example)
+section:
+
+### Mint tokens
+
+The minter mints and transfers NFT tokens for the vaccine program participant.
+In this example, the new participant address is
+`0x2531a4D108619a20ACeE88C4354a50e9aC48ecfe` and the minter address is
+`0x6A33382de9f73B846878a57500d055B981229ac4`.
+
+Please make sure that you are using the same deployer address as a minter
+address, otherwise the `mint` transaction will revert.
+
+```bash
+% truffle console --network aurora
+truffle(aurora)> const cvt = await CovidVaccineToken.deployed()
+truffle(aurora)> await cvt.minter()
+'0x6A33382de9f73B846878a57500d055B981229ac4'
+truffle(aurora)> await cvt.mint('0x2531a4D108619a20ACeE88C4354a50e9aC48ecfe' , {from: accounts[0]})
+```
+
+You should notice that `accounts[0]` is the minter address and none of the
+participants are allowed to transfer their NFT tokens except for the minter.
+
+So let's try to use any participant address to validate this. To do that, change
+the value of `from` to `accounts[1]`, so that the sender will be the first
+participant (e.g., the participant address
+`0x2531a4D108619a20ACeE88C4354a50e9aC48ecfe`).
+
+In the Truffle console:
+
+```bash
+truffle(aurora)> await cvt.safeTransferFrom('0x2531a4d108619a20acee88c4354a50e9ac48ecfe', '0x8722C88e82AbCC639148Ab6128Cd63333B2Ad771', 1, {from: accounts[1]}) 
+Uncaught Error: execution reverted:
+...
+reason: 'Invalid Transfer',
+  hijackedStack: 'Error: execution reverted:\n'
+```
+
+This is exactly the same error message we have in our NFT contract in
+`safeTransferFrom`:
+
+```solidity
+function safeTransferFrom(
+    address from, 
+    address to, 
+    uint256 tokenId
+) 
+    public 
+    virtual 
+    override 
+    {
+        require(
+            minter == msg.sender || to == minter,
+            'Invalid Transfer'
+        );
+        safeTransferFrom(from, to, tokenId, "");
+    }
+```
+
+### Transfer tokens
+
+Participants can transfer the token to the minter after receiving the vaccine.
+As shown below, a participant can only send the NFT token if the receiver for
+this token is the minter (`0x6A33382de9f73B846878a57500d055B981229ac4`).
+
+```bash
+truffle(aurora)> await cvt.ownerOf(1) // TokenID 1
+'0x2531a4D108619a20ACeE88C4354a50e9aC48ecfe'
+truffle(aurora)> await cvt.minter()
+'0x6A33382de9f73B846878a57500d055B981229ac4'
+truffle(aurora)> await cvt.safeTransferFrom('0x2531a4D108619a20ACeE88C4354a50e9aC48ecfe', '0x6A33382de9f73B846878a57500d055B981229ac4', 1 , {from: accounts[1]})
+truffle(aurora)> await cvt.ownerOf(1) // TokenID 1
+'0x6A33382de9f73B846878a57500d055B981229ac4'
+```
+
+### Burn tokens
+
+This is an alternative scenario for the NFT token lifecycle. Instead of
+transfering the token back to the minter, the participant can decide to burn the
+NFT token by calling the `burn` function:
+
+```bash
+truffle(aurora)> await cvt.burn(1, {from: accounts[1]}) // 1 is the tokenID
```

Comment:
Is it a bug that a non-owner of the token can burn? When I was going through the tutorial I was able to burn the token with `accounts[1]` even after I had transferred ownership back to `accounts[0]`.